packages/perseus/src/widgets/group.jsx

```diff
@@ -81,15 +137,12 @@ class Group extends React.Component<Props> {
         // alert our renderer (our parent) of the fact that some interaction
         // has occurred.
         const onInteractWithWidget = (id) => {
-            // eslint-disable-next-line react/no-string-refs
-            if (this.refs.renderer) {
-                // eslint-disable-next-line react/no-string-refs
-                this.change("widgets", this.refs.renderer.props.widgets);
+            if (this.rendererRef) {
+                this.change("widgets", this.rendererRef.props.widgets);
             }
         };
 
-        // eslint-disable-next-line react/no-string-refs
-        const score = this.refs.renderer && this.refs.renderer.score();
+        const score = this.rendererRef?.score();
```

Comment:
Question (non-blocking): The original assignment to this variable looks like it should resolve to a boolean value, but is that how this new assignment will work? If this.rendererRef does not exist, it will assign score null, right? Looking at isValid/isInvalid's logical operators and the return though, maybe this still works?

Reply:
It's essentially the same logic. `refs.renderer` and `rendererRef` (same thing renamed) is a reference to a React component. Both logic is basically shorthand for:

``` JS
let score
// if we try to call `.score()` on undefined we'll get an exception
if (typeof this.rendererRef !== "undefined") {
    score = this.rendererRef.score()
}
```

`const score = this.refs.renderer && this.refs.renderer.score();`

1. We expect `this.refs.renderer` to be a ref or unassigned (`undefined`)
2. If it's assigned it will be truthy, if it's unassigned it will be falsy
3. The `&&` operator returns the first thing if it's falsy and the second thing if the first thing is truthy
4. So if `this.refs.renderer` is unassigned `score` will be assigned `undefined`; otherwise `score` will be assigned the result of `this.refs.renderer.score()`

[Optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) works basically the same (with subtle differences that aren't relevant here).

`const score = this.rendererRef?.score();`

1. We expect `this.rendererRef` to be a ref or unassigned (`undefined`)
2. The `?` operator returns `undefined` if the part before is `undefined` or `null` and the part after otherwise
3. So if `this.rendererRef` is unassigned `score` will be assigned `undefined`; otherwise `score` will be assigned the result of `this.rendererRef.score();`